feat: add intelij LSP integration plugin

bratorange opened the pull request:
See #1744 for more details

bratorange committed:
feat(intellij): Initial IntelliJ plugin structure and LSP setup

bratorange committed:
chore(intellij): Add .gitignore for IntelliJ plugin

bratorange committed:
chore(intellij): Remove build and sandbox dirs from git index

bratorange committed:
fix(intellij): Address build verification warnings and LSP dependency

bratorange committed:
refactor the build files

bratorange committed:
update gitignore

bratorange committed:
[got linting working]

bratorange committed:
feat(intellij): Update PLUGIN_DEV_NOTES and add TinymistInitializationOptions class

- Revised PLUGIN_DEV_NOTES to reflect the current status and next steps for the Tinymist language server integration, including completion of Phase 1 and Phase 2.
- Introduced TinymistInitializationOptions class to structure initialization options for the language server, mirroring VSCode configuration.
- Enhanced TinymistLspStreamConnectionProvider to utilize the new initialization options structure.

bratorange committed:
feat(intellij): Add comprehensive development notes and preview functionality

- Introduced a new development notes file detailing the project scope, current status, next steps, and identified technical debt for the Tinymist IntelliJ plugin.
- Implemented TypstPreviewFileEditor and TypstTextEditorWithPreviewProvider to enable a split view for Typst files, integrating a JCEF-based preview panel.
- Updated plugin.xml to register the new preview editor and removed the previously planned tool window for Typst preview.
- Cleaned up unused code in TypstLexerAdapter and ensured proper handling of the preview functionality.

bratorange committed:
feat(intellij): Enhance Typst preview functionality and add platform-impl dependency; doesnt build yet

- Added an explicit dependency on `platform-impl` in the build configuration to ensure proper functionality.
- Updated `TypstPreviewFileEditor` to improve the loading of the preview page with local assets, including a JavaScript bridge for communication between the IntelliJ plugin and the preview content.
- Enhanced error handling and logging for better debugging during the preview loading process.
- Cleaned up development notes to reflect the current state and future plans for the Typst integration.

bratorange committed:
feat(intellij): Refactor Typst preview integration and remove unused components

- Updated `TypstPreviewFileEditor` to directly load the Tinymist preview server URL, enhancing the integration with the background preview functionality.
- Removed the `TypstPreviewResourceHandler` and related resource handling code, as the preview server now serves all necessary assets.
- Cleaned up the `TinymistLspStreamConnectionProvider` to streamline initialization options and command execution.
- Updated development notes to reflect changes in the preview strategy and current project status.
- Adjusted `.gitignore` to include new entries for IntelliJ platform files.

bratorange committed:
feat(intellij): Improve TypstPreviewFileEditor with JCEF integration and error handling

- Refactored `TypstPreviewFileEditor` to extend `JCEFHtmlPanel`, enhancing the integration with the JCEF browser.
- Implemented a new loading mechanism for the Tinymist preview server, including improved error handling and user feedback.
- Added a label to inform users when JCEF is not supported in their environment.
- Updated `.gitignore` to exclude Kotlin-related files.

bratorange committed:
debug: attempt to debug a performance issue wtih the frontend

- Modified `.cursorignore` to comment out the IntelliJ community directory for better indexing.
- Updated `Cargo.lock` and `Cargo.toml` to reflect changes in the `tinymist-assets` dependency path.
- Added new IntelliJ project configuration files, including `.gitignore`, `compiler.xml`, `encodings.xml`, and others to enhance project setup.
- Enhanced `TypstPreviewFileEditor` with additional logging for better debugging and performance tracking.
- Introduced a new `trace.json` file for performance analysis of the preview functionality.
- Updated development notes to include current focus on preview panel scrolling performance and ongoing investigations.

bratorange committed:
refactor: Update development notes and project roadmap for Typst IntelliJ integration

- Consolidated project status and next steps into a structured roadmap format.
- Documented completed milestones, current focus on preview panel performance, and identified technical debt.
- Clarified the role of the IntelliJ plugin in the preview architecture and outlined future enhancements, including settings panel implementation and robust executable handling.
- Removed outdated sections and streamlined content for better clarity and organization.

bratorange committed:
docs: Update development notes with comprehensive instructions and project roadmap for Tinymist IntelliJ plugin

bratorange committed:
removed accidentally tracked example repo

bratorange committed:
Cleanup: Remove dead code, restore minimal preview provider, and ensure consistency

- Removed dead code and placeholder logic (OutlineDataHolder, mock outline, unused updateOutline).
- Deleted and then restored a minimal TypstPreviewFileEditorProvider to serve as the preview editor provider.
- Updated TypstTextEditorWithPreviewProvider to use the correct preview provider.
- Cleaned up plugin.xml by removing commented-out tool window registration.
- Ensured all code is consistent with dev-notes and current architecture.

bratorange committed:
add mockdata structure view and fix languageclient file missing

bratorange committed:
feat(intellij): Enhance Typst preview and add Find Usages provider

- Added a new `TypstFindUsagesProvider` to support finding usages of named elements in Typst files, delegating scanning to the LSP.
- Improved `TypstPreviewFileEditor` by ensuring the display handler is set up correctly and logging JavaScript console messages for better debugging.
- Updated development notes with detailed insights into frontend performance investigations and ongoing debugging efforts related to the JCEF-based preview panel.

bratorange committed:
refactor(intellij): reset TypstPreviewFileEditor

bratorange committed:
chore(intellij): remove  lexer, parser, syntax highlighter, and structure view components as they are allready done by lps4ij

- Deleted `TypstLexerAdapter`, `TypstParserDefinition`, `TypstSyntaxHighlighter`, and related structure view classes to clean up the codebase.
- These components were no longer necessary following recent refactoring and enhancements to the Typst integration.

bratorange committed:
chore(intellij): update lsp4ij dependency and remove custom implementations which are allready done by lsp4ij

- Updated the `lsp4ij` plugin dependency to version `0.13.0`.
- Added a new lexer definition for Typst, improving the language parsing capabilities.
- Introduced a new BNF grammar file for Typst, establishing a foundation for syntax parsing.
- Removed outdated `TinymistOutlineModel` and `TypstFindUsagesProvider` classes to streamline the codebase.
- Updated development notes to reflect changes in the architecture and language support.

bratorange committed:
chore(intellij): update development notes

bratorange committed:
implement server change feature

bratorange committed:
fix cleanup issue in the preview

bratorange committed:
refactor(TypstPreview): restrict editor to Typst files and enhance selection handling

bratorange committed:
Add tests for Typst LSP functionality

Introduce unit tests to validate Typst plugin LSP features, including "Go to Definition," completion, hover, file type association, and editor preview provider behavior. These tests ensure compatibility with Typst files and verify integration with the LSP server where applicable.

bratorange committed:
added icons

bratorange committed:
Refactor editor to track and manage preview registrations.

bratorange committed:
Refactor test logic and simplify assertion handling.

Removed unnecessary conditional checks and comments related to LSP server behavior in test cases. Consolidated debug logs and assertions for clarity and consistency across all test files.

bratorange committed:
Refactor and remove unused code and development fallbacks.

bratorange committed:
Reduce max server connection attempts from 60 to 5

bratorange committed:
removed accidentally added references to other repos from the repo

bratorange committed:
update dependency lsp4ij to 0.15

bratorange commented:
okay it seems as some slop wasnt cleaned up before. I gone take care of that

bratorange commented:
okay I removed some changes to files which I only did for debugging purposes from the history.

Myriad-Dreamin reviewed (commented):
I left some comments that I observed with only opening the intellij plugins with idea. I have not run the plugin yet.

Myriad-Dreamin commented on editors/intellij/dev-notes.md:
````diff
@@ -0,0 +1,242 @@
+    *   Wait for Gradle sync to complete and download all dependencies.
+    *   You can build the plugin using the Gradle tool window in IntelliJ (Tasks > intellij > buildPlugin) or via the terminal:
+        ```bash
+        ./gradlew buildPlugin
````
We could add a CI to ensure build

bratorange committed:
Delete editors/intellij/trace.json

bratorange commented on editors/intellij/src/main/kotlin/org/tinymist/intellij/lsp/TinymistLspStreamConnectionProvider.kt:
```diff
@@ -0,0 +1,94 @@
+        } ?: LOG.error("Tinymist LSP server commands not set as no executable was found.")
+    }
+
+    private fun findExecutableOnPath(name: String): String? {
```
@gareth s approach seemed to do that pretty well so thought about just using his code here, but am not sure if he is okay with that, so im waiting on his answer. However in the koltin / java std libs I have only found methods for directly invoking commands, not for PATH binary discovery.

bratorange committed:
refactor: adopt descriptive comment style

bratorange committed:
refactor: adopt descriptive comment style

bratorange committed:
fix: remove wrong @Test annotations

bratorange committed:
fix TypstCompletionTest.kt

Myriad-Dreamin commented:
from recent news cc by some of my friend, the lsp plugin will be available for free in 2025.2. see: https://platform.jetbrains.com/t/lsp-api-available-in-intellij-idea-ultimate-2025-2-without-paid-subscription/2284/2

bratorange committed:
add .idea for the intelij plugin folder

Myriad-Dreamin commented:
> > from recent news cc by some of my friend, the lsp plugin will be available for free in 2025.2. see: https://platform.jetbrains.com/t/lsp-api-available-in-intellij-idea-ultimate-2025-2-without-paid-subscription/2284/2
> 
> oh interesting. that would constitute that it probably makes more sense to merge the implementation of @garetht into this repo right?

Not in a rush. Probably a good pace is preparing and moving in the start of 2026 or end of 2025 because many users will be still using 2025.1 in the short term I think. I haven't checked but we can merge the two implementations and then consider combination and migrations.

bratorange committed:
update development info

Myriad-Dreamin commented:
> https://blog.jetbrains.com/zh-hans/idea/2025/07/intellij-idea-unified-distribution-plan/

<img width="1600" height="937" alt="image" src="https://github.com/user-attachments/assets/97e066dc-0587-40e4-89e9-d8735cec749e" />

Myriad-Dreamin commented:
@angelozerr could have ideas or suggestions for selecting a future-proof LSP framework.

Myriad-Dreamin commented on editors/intellij/src/main/kotlin/org/tinymist/intellij/lsp/TinymistLspStreamConnectionProvider.kt:
```diff
@@ -0,0 +1,88 @@
+import org.tinymist.intellij.settings.TinymistSettingsService
+import java.io.File
+
+class TinymistLspStreamConnectionProvider(private val project: Project) : ProcessStreamConnectionProvider() {
```
Fwiw, there are generated `tinymist-installer.{sh,ps1}` on releases https://github.com/Myriad-Dreamin/tinymist/releases/tag/v0.13.24.

bratorange replied:
okay I (my LLM) implemented this via editors/intellij/src/main/kotlin/org/tinymist/intellij/lsp/TinymistLanguageServerInstaller.kt. My idea here was to just use github releases based on a hard coded version Identifier.

bratorange commented on editors/intellij/src/main/kotlin/org/tinymist/intellij/lsp/TinymistLanguageClient.kt:
```diff
@@ -0,0 +1,81 @@
+    override fun showMessageRequest(params: ShowMessageRequestParams): CompletableFuture<MessageActionItem> {
+        LOG.warn("Received showMessageRequest from server. Type: ${params.type}, Message: '${params.message}'. Actions: ${params.actions}. Suppressing UI and returning null action item to avoid potential NPE in lsp4ij.")
+        // Returning a completed future with null, effectively ignoring the request from UI perspective
+        // and preventing the lsp4ij default handler from causing an NPE if params.getActions() is null.
```
The message in question which produces the problem, which necessitate this workaround is the following:

[Trace - 20:55:20] Received request 'window/showMessageRequest - (0)'
Params: {
  "type": 2,
  "message": "bad server configuration: failed to deserialize \"\"semanticTokens\": Error(\"invalid value: map, expected map with a single key\", line: 0, column: 0)"
}

As far as I see this, this could be fixed in three ways:
- Prevent this error message from occouring altogether
- Make LSP4IJ tolerate showMessageRequest messages without an actions attribute
- Change the tinymist lsp server from not serving messages without actions attribute

According to the specification, actions can be null.
```
actions?: MessageActionItem[];
```

I will create an issue in lsp4ij because because a params.action should not be assumed to be non-null here: 
https://github.com/redhat-developer/lsp4ij/blob/d625e089b6eee21bf2db9d47783fe112f030d680/src/main/java/com/redhat/devtools/lsp4ij/ServerMessageHandler.java#L123

Also I need to look in, why the message appears altogether.

bratorange replied:
https://github.com/redhat-developer/lsp4ij/issues/1243

angelozerr replied:
Please install https://github.com/redhat-developer/lsp4ij?tab=readme-ov-file#testing-nightly-builds the bug should be fixed

bratorange replied:
yeah thats another topic. unfortunately the draft still prododuces the bug for me, as all lines will be shown in blue on hover.
<img width="1393" height="712" alt="Screenshot 2025-09-01 at 16 50 01" src="https://github.com/user-attachments/assets/83a4a48b-a704-4b37-92f5-e8e60ff97fee" />
#1178

steps which I took:
1. clone repo from the pr and checkout the branch
2. build with ./gradlew buildPlugin
3. replace this in my build.gradle.kts
```
    intellijPlatform {
        intellijIdeaUltimate("2025.1")
        testFramework(TestFrameworkType.Platform)
        localPlugin(file("/Users/juliusschmitt/kotlin/lsp4ij/build/distributions/lsp4ij-0.14.3-SNAPSHOT.zip"))
    }
```

bratorange replied:
This is the output of the semantic token inspector:
```
<null>=</null><null> </null><null><n</null>ull>H1</null><null>

</null><null>==</null><null> </null><null><n</null>ull>h2</null><null>
</null><null>content</null><null>

    </null><null>[</null><null>

    </null><null>()</null><null>

</null><null>]</null><null>

</null><LSP_KEYWORD>#</LSP_KEYWORD><LSP_KEYWORD>let</LSP_KEYWORD><null> </null><null>test</null><null> </null><LSP_OPERATOR>=</LSP_OPERATOR><null> </null><LSP_STRING>"test"</LSP_STRING><null>
</null><null>#</null><null>test</null><null>
</null><null>$</null><null>x</null><null> </null><null>=</null><null> </null><null>500</null><null>$</null>
```

So I guess the fact all of the types are null could probably interfere with the PSI generation?

bratorange committed:
feature: tinymist binary installer

bratorange committed:
refactor: centralize Tinymist version management and adapt server customization options to the new server initializer

bratorange committed:
refactor: replace ProcessStreamConnectionProvider with OSProcessStreamConnectionProvider

bratorange committed:
refactor: add custom diagnostic feature

- Introduced `TinymistLSPDiagnosticFeature` to handle multi-line diagnostic messages.
- Updated `TinymistLanguageServerFactory` to use the custom diagnostic feature.
- Removed redundant `publishDiagnostics` override from `TinymistLanguageClient`.

bratorange commented on editors/intellij/src/main/kotlin/org/tinymist/intellij/lsp/TinymistLanguageClient.kt:
```diff
@@ -0,0 +1,81 @@
+        private val LOG = Logger.getInstance(TinymistLanguageClient::class.java)
+    }
+
+    override fun publishDiagnostics(diagnostics: PublishDiagnosticsParams) {
```
I applied this suggestion via TinymistLSPDiagnosticFeature.kt. Thanks for the advice!

bratorange committed:
refactor: dont use findExecutableOnPath anymore

angelozerr commented on editors/intellij/src/main/kotlin/org/tinymist/intellij/lsp/TinymistLanguageServerInstaller.kt:
```diff
@@ -0,0 +1,294 @@
+ * Downloads and installs the Tinymist binary based on the current platform.
+ * Uses the correct GitHub asset names from the actual releases.
+ */
+class TinymistLanguageServerInstaller : LanguageServerInstallerBase() {
```
IMHO, I think you should use the declarative installer from LSP4IJ because:

 * with just a JSON file you can describe the installation
 * donwload files is robust and manages for instance progress of the installation.

To load a JSON installer, I suggest that you do like https://github.com/redhat-developer/lsp4ij/blob/main/src/main/java/com/redhat/devtools/lsp4ij/server/definition/launching/UserDefinedLanguageServerInstaller.java

angelozerr replied:
To write JSON installer, I suggest that you do like rust-analyzer wich uses Githu assets https://github.com/redhat-developer/lsp4ij/blob/main/src/main/resources/templates/lsp/rust-analyzer/installer.json

bratorange committed:
remove unnecessary implementation of LanguageServerImpl

bratorange committed:
chore: remove unused grammar definitions

bratorange committed:
fixed multiple versions of gradle being referenced?

bratorange committed:
reference a nightly build of lsp4ij

bratorange commented on editors/intellij/gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -0,0 +1,7 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-bin.zip
```
I removed the conflgicting version declaration in [823ef6f](https://github.com/Myriad-Dreamin/tinymist/pull/2087/commits/823ef6f53c0077c07517524e4fce35ebc4dc31f7)

bratorange committed:
restored textDocument/foldingRange ability

bratorange committed:
update dev-notes.md to reflect current state

bratorange committed:
update dev-notes.md to reflect current state

bratorange committed:
Merge remote-tracking branch 'origin/main'

bratorange committed:
update dev prerequisites

bratorange committed:
readd preview init options

bratorange committed:
add top level build file

bratorange committed:
implement preview port collision prevention

Myriad-Dreamin committed:
dev: add Gradle wrapper and configuration files

Myriad-Dreamin committed:
dev: update IntelliJ Gradle configuration to use Community edition

Myriad-Dreamin committed:
docs: edit development guides

Myriad-Dreamin commented:
I move and edit the dev-notes. The bigest change is that we remove these two steps because we develop the intellij plugin from the root directory:

```
2.  **Clone the Repository:**
    ```bash
    git clone https://github.com/Myriad-Dreamin/tinymist.git
    cd tinymist/editors/intellij
    ```
3.  **Open in IntelliJ IDEA:**
    *   Open IntelliJ IDEA.
    *   Select "Open" and choose the root directory of the cloned repository.
    *   IntelliJ should automatically recognize Gradle projects. If not, you might need to import it as a Gradle project.
```

I mainly edit the development instructions, and leave rest part untouched. TBH, the rest parts may be summarized by LLM-powered search engine, for example, deep-wiki. I'm okay if we maintain an overview for people.

Myriad-Dreamin committed:
docs: fix link to Myriad-Dreamin's VS Code settings in development guide

bratorange committed:
update build instruction to use community edition

angelozerr commented on docs/dev-guide/intellij-plugin.md:
```diff
@@ -0,0 +1,187 @@
+| `textDocument/documentSymbol`              | ✅ Implemented        | Handled by lsp4ij     | Document outline/structure view                                                                                                  |
+| `textDocument/inlayHint`                   | ✅ Implemented        | Handled by lsp4ij     | Inlay additional information into code editor, i.e. the names of function parameters                                             |
+| `textDocument/codeAction`                  | ✅ Implemented        | Handled by lsp4ij     | Code fixes and refactoring actions                                                                                               |
+| `textDocument/formatting`                  | ❌ Not implemented    | -                     | Document formatting                                                                                                              |
```
Lsp4j can support it, why you  say not implemented ?

angelozerr commented on docs/dev-guide/intellij-plugin.md:
```diff
@@ -0,0 +1,187 @@
+| `textDocument/inlayHint`                   | ✅ Implemented        | Handled by lsp4ij     | Inlay additional information into code editor, i.e. the names of function parameters                                             |
+| `textDocument/codeAction`                  | ✅ Implemented        | Handled by lsp4ij     | Code fixes and refactoring actions                                                                                               |
+| `textDocument/formatting`                  | ❌ Not implemented    | -                     | Document formatting                                                                                                              |
+| `textDocument/rangeFormatting`             | ❌ Not implemented    | -                     | Range-based formatting                                                                                                           |
```
Lsp4j can support it, why you  say not implemented ?

angelozerr commented on docs/dev-guide/intellij-plugin.md:
```diff
@@ -0,0 +1,187 @@
+| `textDocument/codeAction`                  | ✅ Implemented        | Handled by lsp4ij     | Code fixes and refactoring actions                                                                                               |
+| `textDocument/formatting`                  | ❌ Not implemented    | -                     | Document formatting                                                                                                              |
+| `textDocument/rangeFormatting`             | ❌ Not implemented    | -                     | Range-based formatting                                                                                                           |
+| `textDocument/onTypeFormatting`            | ❌ Not implemented    | -                     | Format-on-type                                                                                                                   |
```
Lsp4j can support it, why you  say not implemented ?

bratorange replied:
I have tested it and it did not work, however im not sure if tinymist supports it all, which is why I have left the note as unimplemented

angelozerr commented on docs/dev-guide/intellij-plugin.md:
```diff
@@ -0,0 +1,187 @@
+| `textDocument/formatting`                  | ❌ Not implemented    | -                     | Document formatting                                                                                                              |
+| `textDocument/rangeFormatting`             | ❌ Not implemented    | -                     | Range-based formatting                                                                                                           |
+| `textDocument/onTypeFormatting`            | ❌ Not implemented    | -                     | Format-on-type                                                                                                                   |
+| `textDocument/codeLens`                    | ❌ Not implemented    | -                     | Inline code annotations                                                                                                          |
```
Lsp4j can support it, why you  say not implemented ?

bratorange replied:
I will also need to customly implement code lens actions, which I have not done yet.

angelozerr replied:
Indeed but codelens should be displayed, right?

angelozerr commented on docs/dev-guide/intellij-plugin.md:
```diff
@@ -0,0 +1,187 @@
+| `textDocument/codeLens`                    | ❌ Not implemented    | -                     | Inline code annotations                                                                                                          |
+| `textDocument/foldingRange`                | ✅ Implemented        | Handled by lsp4ij     | Code folding regions                                                                                                             |
+| `textDocument/selectionRange`              | ✅ Implemented        | Handled by lsp4ij     | Smart text selection                                                                                                             |
+| `textDocument/prepareCallHierarchy`        | ❌ Not implemented    | -                     | Call hierarchy preparation                                                                                                       |
```
Lsp4j can support it, why you  say not implemented ?

bratorange replied:
prepareCallHierarchy also did not work in my experiments, however im not sure, where its failing

Myriad-Dreamin commented:
Since I have run it successfully on my computer, I believe this PR has made some significant process. It is not a problem if there are still some bugs or bad smells in current implementation, instead the only missing is to add a CI to ensure the plugin to build. We should merge it once a simple CI is built. The CI must produce some artifacts to help people use the plugin without setting up build env locally. I don't know what specific artifacts to produce, but may be some bundle like `*.vsix` for vscode in current CI.

- [ ] write a CI script to generate artifacts.
- [ ] add a file `frontend/intellij.md` in https://github.com/Myriad-Dreamin/tinymist/tree/main/docs/tinymist/frontend, and describe the instructions to install the plugin from marketplace or CI.
- [ ] ping me and we finally verify that people can get intellij integration by following the instructions.
- [ ] merge the PR.

bratorange committed:
remove an accidental item from plugin.xml

bratorange commented:
> Since I have run it successfully on my computer, I believe this PR has made some significant process. It is not a problem if there are still some bugs or bad smells in current implementation, instead the only missing is to add a CI to ensure the plugin to build. We should merge it once a simple CI is built. The CI must produce some artifacts to help people use the plugin without building locally. I don't know what specific artifacts to produce, but may be some bundle like `*.vsix` for vscode in current CI.
> 
>     * [ ]  write a CI script to generate artifacts.
> 
>     * [ ]  add a file `frontend/intellij.md` in https://github.com/Myriad-Dreamin/tinymist/tree/main/docs/tinymist/frontend, and describe the instructions to install the plugin from marketplace or CI.
> 
>     * [ ]  ping me and we finally verify that people can get intellij integration by following the instructions.
> 
>     * [ ]  merge the PR.

Great to hear! Im going to work on the CI.

bratorange committed:
add github action for building

bratorange committed:
use the original template ci file

bratorange committed:
fix gradlew

bratorange committed:
fix wrong artifact path; update to lsp4ij 0.16.1

bratorange committed:
fix wrong artifact path again

bratorange committed:
Add plugin metadata

Myriad-Dreamin commented:
> regarding the plugin logo: I found an icon called tinymist.png. However intellij demands an svg. Do you still have that?

![ti](https://github.com/user-attachments/assets/399635e4-8b2f-4281-ab65-f2e4c78f6d8b)

bratorange committed:
add plugin Icon

bratorange committed:
add plugin Icon dark

bratorange committed:
add short description of the plugin

bratorange committed:
simplify intellij build action

bratorange committed:
update gradle and lsp4ij

bratorange committed:
update documentatiton; remove unneded functions for the future

bratorange committed:
remove the nececity to verify the plugin to produce a release candidate for now

bratorange committed:
fix changelog export in the ci

bratorange committed:
attempt to fix "open ./build/tmp/release_note.txt: no such file or directory"

bratorange committed:
update intellij platform

bratorange committed:
dont assume any path structure for the changelog file

bratorange committed:
hopefully the release notes path works this time

Myriad-Dreamin committed:
ci: resolve main merge conflicts

Myriad-Dreamin committed:
Merge remote-tracking branch 'origin/main' into bratorange/main

# Conflicts:
#	.github/workflows/ci.yml
#	.gitignore

Myriad-Dreamin committed:
ci: allow manual intellij workflow runs

Myriad-Dreamin committed:
ci: skip intellij release draft on manual dispatch

Myriad-Dreamin committed:
ci: stabilize intellij plugin checks

Myriad-Dreamin committed:
ci: support older lsp4ij verifier APIs

Myriad-Dreamin committed:
ci: fix intellij workflow artifact paths

Myriad-Dreamin committed:
Merge origin/main into bratorange/main

Myriad-Dreamin committed:
ci: configure tinymist for intellij tests

Myriad-Dreamin commented:
Thanks for contribution. I fixed CI and will merge. I currently have no plan to publish this plugin on jetbrains marketplace, but it will be available in artfacts of each CI github action and GitHub release.
